src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -43,6 +43,10 @@ public override Task SendAllAsync(string methodName, object[] args, Cancellation
                 throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(methodName));
             }
 
+            //return ServiceConnectionContainer.WriteAsync(
+            //    new BroadcastDataMessage(null, SerializeAllProtocols(methodName, args), 
+            //    tracingId: ServiceConnectionContainerScope.ServiceDiagnosticLogsContext.IsEnableMessageLog ? "id": null));
```

Comment:
ServiceDiagnosticLogsContext?.

Reply:
Look like if `ServiceConnectionContainerScope.ServiceDiagnosticLogsContext == null` there must be a bug, since the scope is created in [StartAsync()](https://github.com/Azure/azure-signalr/blob/7b627a95c68eb59fae4ed898f1d2ab0d90c8a282/src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs#L148). So, still `ServiceDiagnosticLogsContext.xxx`?

Reply:
This will be null if a customer (or the library they use) suppresses execution context flow. 

I am also concerned that the execution context can be swapped pretty easily.
For example user code can have a timer that periodically dispatches signalr messages to different endpoints (some have tracing enabled some don't). Whoever creates this timer will make every message sent from within this timer carry the tracing/logging/diagnostic flag for every endpoint that may have nothing to do with the original endpoint tracing settings. While the services that don't have tracing enabled will not trace anything, the SDK side will trace for all the messages originated from this timer (which can further confuse the users).

Reply:
> This will be null if a customer (or the library they use) suppresses execution context flow.

Right, make sense.

> I am also concerned that the execution context can be swapped pretty easily.
For example user code can have a timer that periodically dispatches signalr messages to different endpoints (some have tracing enabled some don't). Whoever creates this timer will make every message sent from within this timer carry the tracing/logging/diagnostic flag for every endpoint that may have nothing to do with the original endpoint tracing settings. While the services that don't have tracing enabled will not trace anything, the SDK side will trace for all the messages originated from this timer (which can further confuse the users).

I don't think the users will be confused. The original endpoint's tracing setting have control of the message inside server, but won't impact other endpoints. And the tracing setting will be updated if the service send a ping to notify. I don't see why it's relevant to timer... Maybe I misunderstand?

Reply:
look like your concern is similar to [this](https://github.com/Azure/azure-signalr/pull/923#discussion_r449967890).

Reply:
So if the original endpoint's tracing is on - is it expected that the app server will trace outcoming messages to other endpoints too?

Reply:
no, unless other endpoints also enable message log

Reply:
Then there are many scenarios where a cross-leaked execution context (and the tracing flag it carries) can break this assumption. One such case is the timer example above, but there are really many-many others.